nilearn/input_data/nifti_labels_masker.py

```diff
@@ -273,6 +275,14 @@ def transform_single_imgs(self, imgs, confounds=None):
         # We handle the resampling of labels separately because the affine of
         # the labels image should not impact the extraction of the signal.
 
+        # If a 3D image is given, an extra dimension will be added
+        is_3d = False
+        
+        if type(imgs) != list:
+            if len(imgs.shape) < 4:
```

Comment:
`imgs` can also be a string in case we pass a filename in. In this case, this will fail because strings have no `shape` argument.

Reply:
Hello nicolas,
In fact this was my first PR before discovering the fact of some local tests that i can use before committing the changes. Therefore, i did another PR while adding an extra parameter called is_3d to be indicated. I still have a small issue to check before submitting my last changes